src/Dialect/Mlir/DialectBuilder.hpp

```diff
@@ -259,6 +259,84 @@ struct GenericAffineBuilder final : DialectBuilder {
 using AffineBuilder =
     GenericAffineBuilder<mlir::AffineLoadOp, mlir::AffineStoreOp>;
 
+//===----------------------------------------------------------------------===//
+// LLVM Builder
+//===----------------------------------------------------------------------===//
+
+struct LLVMBuilder final : DialectBuilder {
+  LLVMBuilder(mlir::OpBuilder &b, mlir::Location loc)
+      : DialectBuilder(b, loc) {}
+  LLVMBuilder(DialectBuilder &db) : DialectBuilder(db) {}
+
+  // AddressOfOp
+  mlir::Value addressOf(mlir::LLVM::GlobalOp op) const;
+
+  // AllocaOp
+  mlir::Value _alloca(
+      mlir::Type resultType, mlir::Value size, int64_t alignment) const;
+
+  // BitcastOp
+  mlir::Value bitcast(mlir::Type type, mlir::Value val) const;
+  mlir::Value bitcastI8Ptr(mlir::Value val) const;
+  mlir::Value bitcastI8PtrPtr(mlir::Value val) const;
+
+  // CallOp
+  mlir::Value call(mlir::ArrayRef<mlir::Type> resultTypes,
+      llvm::StringRef funcName, mlir::ArrayRef<mlir::Value> inputs) const;
+  mlir::Value call(mlir::ArrayRef<mlir::Type> resultTypes,
+      mlir::FlatSymbolRefAttr funcSymbol,
+      mlir::ArrayRef<mlir::Value> inputs) const;
+
+  // ConstantOp
+  mlir::Value constant(mlir::Type type, double val) const;
+
+  // ExtractValueOp
+  mlir::Value extractValue(mlir::Type resultType, mlir::Value container,
+      llvm::ArrayRef<int64_t> position) const;
+
+  // FuncOp
```

Comment:
Feel a bit strange to see `mlir::LLVM` as usually LLVM stands on it's own. So you felt the need to put LLVFuncOp under the mlir::LLVM.

Reply:
I guess you mean `llvm` not `LLVM`. LLVM is indeed inside `mlir`.